internal/service/alarms/api/server.go

```diff
@@ -492,14 +492,41 @@ func (a *AlarmsServer) UpdateAlarmServiceConfiguration(ctx context.Context, requ
 
 }
 
-func (a *AlarmsServer) GetProbableCauses(ctx context.Context, request api.GetProbableCausesRequestObject) (api.GetProbableCausesResponseObject, error) {
-	// TODO implement me
-	return nil, fmt.Errorf("not implemented")
+// GetProbableCauses list all the relevant probableCauses based on alarms that are currently in the events table
+func (a *AlarmsServer) GetProbableCauses(ctx context.Context, _ api.GetProbableCausesRequestObject) (api.GetProbableCausesResponseObject, error) {
+	defs, err := a.AlarmsRepository.GetAllAlarmDefForProbableCause(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get alarm definitions for probableCauses: %w", err)
+	}
+
+	pcs := make([]api.ProbableCause, 0, len(defs))
+	for _, def := range defs {
+		pcs = append(pcs, models.ConvertDefToPC(def))
+	}
+
+	slog.Info("Successfully retrieved probableCauses of current events", "count", len(pcs))
+	return api.GetProbableCauses200JSONResponse(pcs), nil
 }
 
+// GetProbableCause retrieve a probable cause of alarm with UUID
 func (a *AlarmsServer) GetProbableCause(ctx context.Context, request api.GetProbableCauseRequestObject) (api.GetProbableCauseResponseObject, error) {
-	// TODO implement me
-	return nil, fmt.Errorf("not implemented")
+	alarmDefs, err := a.AlarmsRepository.GetAlarmDefForProbableCause(ctx, request.ProbableCauseId)
+	if err != nil {
+		return nil, fmt.Errorf("couldnt not retrieve alarms for probable cause: %w", err)
```

Comment:
Maybe use the same error text as above, `failed to get alarm definitions for probableCauses` or similar to the `Detail` from below?
Also, typo: `couldnt not`.

Reply:
updated!